### ISSUE TITLE
[BRX/APL] Encoding crashes GPU when using map instead of map_gtt

### ISSUE THREAD

**Issue by digitaldriver** [intel/media-driver#1255]
gstreamer version: 1.18.4
VA-API version: 1.11 (libva 2.11.0)
Driver version: Intel iHD driver for Intel(R) Gen Graphics - 21.1.3 
CPU: Intel(R) Atom(TM) Processor E3950 @ 1.60GHz

Following the instructions from #1191 I was able to fix the "slow frame fetch from GPU to CPU" when decoding. But when trying to use vaapih264enc to encode, it fails and the GPU crashes. 

gstreamer sample pipeline:
gst-launch-1.0 -v videotestsrc ! video/x-raw,format=NV12,width=1280,height=720,framerate=30/1 ! vaapih264enc ! fakesink

Error I get is along the lines of 
“vaapih264enc failed to allocate encoded buffer in system memory”
and at times
" drm:_intel_uc_reset_hw [i915]] *ERROR* Failed to reset GuC, ret = -110 “

If I revert the changes to use map_gtt, the encoding works fine without any crash. Why does the vaapih264enc fail to allocate memory? 

Thanks in advance.

**Comment by XinfengZhang** [comment]
map_gtt will do the de-swizzle , convert the surface from tile to linear, map not, but it should not crash.  but unfortunately , I have no such platform aside. so may not to reproduce it. , could you help to collect some information, such as LIBVA_TRACE?

**Comment by digitaldriver** [comment]
Thank you for your response. Wil try and get the LIBVA_TRACE. 

Meanwhile I have attached the patch file that I used to get the decoder working. If I revert map to map_gtt in Lock and Unlock funtions(line 67 and 78 as highlighed in pdf file), the GPU doesn't crash any more and I can encode. Although the encoded image(colour bars) is tiled. I have attached that too for your reference. 

Decoder still works as expected with above changes.

[untile-patch.pdf](https://github.com/intel/media-driver/files/7104442/untile-patch.pdf)
![colourBars_encoded](https://user-images.githubusercontent.com/89645683/131975989-1cba99cc-5317-49da-9c0c-d4d2f8bdc163.jpg)
